std/range/primitives.d

```diff
@@ -2192,7 +2192,7 @@ the first argument using the dot notation, `array.save` is
 equivalent to `save(array)`. The function does not duplicate the
 content of the array, it simply returns its argument.
  */
-@property T[] save(T)(T[] a) @safe pure nothrow @nogc
+@property inout(T)[] save(T)(return scope inout(T)[] a) @safe pure nothrow @nogc
```

Comment:
Given that this is already a templated function, what does adding `inout` add here (or anywhere in this entire PR)? Unless I misunderstand what the compiler does, it doesn't even save any template instantiations. it just makes the function signature more complicated for no real benefit. The primary benefit of `inout` is allowing a single function to work with `const`, `immutable`, and mutable when it's _not_ a template.

Reply:
It results in fewer distinct functions in the final executable. See https://github.com/dlang/phobos/pull/6743#issuecomment-435770952